### PR TITLE
Adding additional params that can be passed in the body during token …

### PR DESCRIPTION
**Pull request by sowmiya**
…exchange

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @sowmiya has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m [hunk 1]
```diff
@@ -180,6 +180,12 @@ - (NSString *)redirectURI
     return _redirectURI;
 }
 
+- (void)setAdditionalTokenGrantParams:(NSDictionary *)additionalTokenGrantParams
+{
+     BOXOAuth2Session *OAuth2Session = (BOXOAuth2Session *)self.SDKClient.session;
+    OAuth2Session.additionalTokenGrantParams = additionalTokenGrantParams;
```
Comment (RicoYao) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m:186]: minor: spacing

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXOAuth2Session.m [hunk 2]
```diff
@@ -99,15 +99,15 @@ - (void)performAuthorizationCodeGrantWithReceivedURL:(NSURL *)URL withCompletion
...
-                                 };
-
+    NSMutableDictionary *POSTParams = [NSMutableDictionary dictionaryWithDictionary:@{
+                                                                                      BOXAuthTokenRequestGrantTypeKey : BOXAuthTokenRequestGrantTypeAuthorizationCode,
+                                                                                      BOXAuthTokenRequestAuthorizationCodeKey : authorizationCode,
+                                                                                      BOXAuthTokenRequestClientIDKey : self.clientID,
+                                                                                      BOXAuthTokenRequestClientSecretKey : self.clientSecret,
+                                                                                      BOXAuthTokenRequestRedirectURIKey : self.redirectURIString,
+                                                                                      }];
+    [POSTParams addEntriesFromDictionary:self.additionalTokenGrantParams];
```
Comment (RicoYao) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXOAuth2Session.m]: Just confirming: addEntriesFromDictionary is nil-safe?

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m (hunk 1 above)
Comment (sowmiya) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m:186]: fixed.

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXOAuth2Session.m (hunk 2 above)
Comment (sowmiya) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXOAuth2Session.m]: it is nil safe. But decided to introduce the check for existence of contents within the dictionary before adding.

**Comment by RicoYao** [comment]
+2

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.h [hunk 3]
```diff
@@ -31,6 +31,10 @@
  */
 @interface BOXAuthorizationViewController : UIViewController <UIWebViewDelegate>
 
+/**
+ * Custom parameters to use in the token grant request. Optional, defaults to nil.
+ */
+@property (nonatomic, readwrite, strong) NSDictionary *additionalTokenGrantParams;
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.h:37]: Parameters.

additionalAuthorizationParameters ?

**Comment by boxtcarpel** [comment]
We talked about exposing just in the session, not in the auth VC

**Pull request merged**